app/uk/gov/hmrc/personaldetailsvalidation/PersonalDetailsValidationResourceController.scala

```diff
@@ -70,7 +79,16 @@ object PersonalDetailsValidationResourceController {
     (__ \ "firstName").readOrError[String]("firstName is missing").filter(ValidationError("firstName is blank/empty"))(_.trim.nonEmpty) and
       (__ \ "lastName").readOrError[String]("lastName is missing").filter(ValidationError("lastName is blank/empty"))(_.trim.nonEmpty) and
       (__ \ "dateOfBirth").readOrError[LocalDate]("dateOfBirth is missing/invalid") and
-      (__ \ "nino").readOrError[String]("nino is missing").map(_.toUpperCase.replaceAll("""\s""", "")).filter(ValidationError("invalid nino format"))(str => Try(Nino(str)).isSuccess).map(Nino)
-    ) (PersonalDetails(_, _, _, _))
-
+      (__ \ "nino").readNullable[String].map {
```

Comment:
This READS code can be modified to return either an instance of the ADT structure I mentioned here - https://github.com/hmrc/personal-details-validation/pull/13/files#r171514238